tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,549 @@
-ica.fit(raw, picks='meg', decim=decim, reject=reject)
-print(ica)
+ecg_evoked = create_ecg_epochs(raw).average()
+ecg_evoked.plot_joint()
 
 ###############################################################################
-# Plot ICA components
-ica.plot_components()  # can you spot some potential bad guys?
-
+# Filtering to remove slow drifts
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Before we run the ICA, an important step is filtering the data to remove
+# low-frequency drifts, which can negatively affect the quality of the ICA fit.
+# The slow drifts are problematic because they reduce the independence of the
+# assumed-to-be-independent sources (e.g., during a slow upward drift, the
+# neural, heartbeat, blink, and other muscular sources will all tend to have
+# higher values), making it harder for the algorithm to find an accurate
+# solution. A high-pass filter with 1 Hz cutoff frequency is recommended.
+# However, because filtering is a linear operation, the ICA solution found from
+# the filtered signal can be applied to the unfiltered signal (see [2]_ for
+# more information), so we'll keep a copy of the unfiltered
+# :class:`~mne.io.Raw` object around so we can apply the ICA solution to it
+# later.
+
+filt_raw = raw.copy()
+filt_raw.load_data().filter(l_freq=1., h_freq=None)
 
 ###############################################################################
-# Component properties
-# --------------------
+# Fitting and plotting the ICA solution
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Now we're ready to set up and fit the ICA. Since we know (from observing our
+# raw data) that the EOG and ECG artifacts are fairly strong, we would expect
+# those artifacts to be captured in the first few dimensions of the PCA
+# decomposition that happens before the ICA. Therefore, we probably don't need
+# a huge number of components to do a good job of isolating our artifacts
+# (though it is usually preferable to include more components for a more
+# accurate solution). As a first guess, we'll run ICA with ``n_components=15``
+# (use only the first 15 PCA components to compute the ICA decomposition) — a
+# very small number given that our data has over 300 channels, but with the
+# advantage that it will run quickly and we will able to tell easily whether it
+# worked or not (because we already know what the EOG / ECG artifacts should
+# look like).
 #
-# Let's take a closer look at properties of first three independent components.
+# .. sidebar:: Ignoring the time domain
+#
+#     The ICA algorithms implemented in MNE-Python find patterns across
+#     channels, but ignore the time domain. This means you can compute ICA on
+#     discontinuous :class:`~mne.Epochs` or :class:`~mne.Evoked` objects (not
+#     just continuous :class:`~mne.io.Raw` objects), or only use every Nth
+#     sample by passing the ``decim`` parameter to ``ICA.fit()``.
+#
+# ICA fitting involves some randomness (e.g., the components may get a sign
```

Comment:
maybe a better way to put this is: "ICA fitting will give different solutions for different initializations"? Saying ICA fitting involves randomness gives me the impression that one cannot trust the algorithm